### ISSUE TITLE
Schema generator overwrites explicitly defined url patterns to views with common methods.

### ISSUE THREAD

**Issue by marcgibbons** [tomchristie/django-rest-framework#4463]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [x] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce
- Create 2 API views with a `get` method
- Define urlpatterns for the views: one as a "list", one as a "detail" (i.e. `/snippets/`, `/snippets/{pk}`)
- Generate the schema

## Expected behaviour
Given a "list" and "detail" view with explicitly defined urlpatterns, and that the views support common HTTP methods, Document Link objects should be created for both the "list" and "detail" endpoints.

## Actual behaviour
A link is only created for the "detail" route.

```python
Document(
    url='', 
    title='Test View', 
    content={
        'snippets': {
              'read': Link(
                  url='/snippets/{pk}/', 
                  action='get', 
                  fields=[Field('pk', required=True, location='path')]
              )
         }
     }
)
```

**Comment by tomchristie** [comment]
Thanks Marc.

**Comment by tomchristie** [comment]
Resolved via #4527, which now checks for the list vs read case. Now in the `version-3-5` branch.
